patches/hide-profile-icon-feature.patch

```diff
@@ -0,0 +1,43 @@
+diff --git a/chrome/browser/ui/ui_features.cc b/chrome/browser/ui/ui_features.cc
+index 097362cfceb53..f650be26cc96d 100644
+--- a/chrome/browser/ui/ui_features.cc
++++ b/chrome/browser/ui/ui_features.cc
+@@ -13,6 +13,10 @@
+ 
+ namespace features {
+ 
++// Hides the toolbar profile icon in regular profiles
++BASE_FEATURE(kHideProfileIcon, "HideProfileIcon",
++             base::FEATURE_ENABLED_BY_DEFAULT);
++
+ // Enables the tab dragging fallback when full window dragging is not supported
+ // by the platform (e.g. Wayland). See https://crbug.com/896640
+ BASE_FEATURE(kAllowWindowDragUsingSystemDragDrop,
+diff --git a/chrome/browser/ui/ui_features.h b/chrome/browser/ui/ui_features.h
+index c971ba1b46016..b388a087e96d5 100644
+--- a/chrome/browser/ui/ui_features.h
++++ b/chrome/browser/ui/ui_features.h
+@@ -21,6 +21,8 @@ namespace features {
+ // All features in alphabetical order. The features should be documented
+ // alongside the definition of their values in the .cc file.
+ 
++BASE_DECLARE_FEATURE(kHideProfileIcon);
++
+ // TODO(crbug.com/40598679): Remove this when the tab dragging
+ // interactive_ui_tests pass on Wayland.
+ BASE_DECLARE_FEATURE(kAllowWindowDragUsingSystemDragDrop);
+diff --git a/chrome/browser/ui/views/toolbar/toolbar_view.cc b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+index 05ae08a41e203..206b92bfff2e6 100644
+--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
++++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+@@ -455,7 +455,9 @@ void ToolbarView::Init() {
+ 
+   avatar_ = container_view_->AddChildView(
+       std::make_unique<AvatarToolbarButton>(browser_view_));
+-  bool show_avatar_toolbar_button = true;
++  bool show_avatar_toolbar_button =
++      !(browser_->profile()->IsRegularProfile() &&
```

Comment:
what is this check doing? This would mean that if hideProfileIcon is disabled but the browser profile isn't a regular profile, we won't show the icon. why is this desirable?

Reply:
nvm i misread the parens. so this will only be true if the profile isn't regular *or* the feature to hide the profile icons is disabled

This makes more sense but I'm still unclear on the need to check whether the profile is regular

Reply:
@RKNF404 hmm so basically today the profile icon is always shown, but with this we'll disable it for incognito and guest even if the flag isn't set. 

This seems like a bit of an aggressive design change as opposed to changing the color or something.

couldn't we just have:

bool show_avatar_toolbar_button = !base::FeatureList::IsEnabled(features::kHideProfileIcon);

Reply:
Oh I see, that was my gap. I didn't realize the intent was to always show for incognito and guest. Ok this looks good

Reply:
Thanks for the refresher on De Morgan's laws :P

I wasn't clear on the intent but it makes sense now, approved